lib/src/generator_database.dart

```diff
@@ -138,4 +140,119 @@ class GeneratorDatabase {
           (e) => _isKind(e, KnownTypeKind.attribute),
         );
   }
+
+  final _allPropsCache = HashMap<String, List<PropertyRef>>();
+
+  /// Follows the inheritance tree to retrieve all properties of the given statement.
```

Comment:
Be wary of treating this as an inheritance - it is not. It is only `extension` on the contract level (i.e. you don't inherit anything, you just extend a base types with more properties).

Reply:
What would be a more appropriate name for it? Extension tree? Structural tree?

Reply:
I would personally go with a variation of `extend`/`extension` or something like that.